### PR TITLE
Add features of gorequests and some helper functions

### PR DESCRIPTION
**Pull request by zhouzhuojie**
Add some helper functions, and also fixed `send_http` by using a new package - `gorequest`

```go
func genLocalFuncMap(om *OpenMock) template.FuncMap {             
    return template.FuncMap{                                      
        "htmlEscapeString":       template.HTMLEscapeString,      
        "isLastIndex":            isLastIndex,                    
        "jsonPath":               jsonPath,                       
        "redisDo":                redisDo(om),                    
        "regexFindAllSubmatch":   regexFindAllSubmatch,           
        "regexFindFirstSubmatch": regexFindFirstSubmatch,         
        "uuidv5":                 uuidv5,                         
        "xmlPath":                xmlPath,                        
    }                                                             
}                                                                 
```

### CODE REVIEW

File: template_helper_test.go [hunk 1]
```diff
@@ -56,35 +56,58 @@ func TestHelpers(t *testing.T) {
...
 
-func TestRedisDo(t *testing.T) {
-	om := &OpenMock{}
-	r := redisDo(om)
+	t.Run("regexFindFirstSubmatch helpers", func(t *testing.T) {
+		raw := `{{ "peach" | regexFindFirstSubmatch "p([a-z]+)ch" }}`
+		c := &Context{}
+		r, err := c.Render(raw)
+		assert.NoError(t, err)
+		assert.Equal(t, r, "ea")
```
Comment (zhouzhuojie) [inline template_helper_test.go]: @ziru fyi

File: demo_templates/http.yaml [hunk 2]
```diff
@@ -54,3 +54,24 @@
...
+          Content-Type: text/xml
+        body: >
+          <xml>
+            {{` <note>
+                  <to>Tove</to>
+                  <from>Jani</from>
+                  <heading name=\"heading\">Reminder</heading>
+                  <body>Don't forget me this weekend!</body>
+                </note>
+              ` | htmlEscapeString }}
```
Comment (zhouzhuojie) [inline demo_templates/http.yaml]: example of doing `htmlEscapeString`

File: demo_templates/regex.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      path: /regex
+  actions:
+    - reply_http:
+        status_code: 200
+        body: >
+          {
+            "match first string": "{{.HTTPBody | regexFind "peach"}}",
+            "match and get first group": "{{.HTTPBody | regexFindFirstSubmatch "p([a-z]+)ch" }}",
+            "match and get all groups": "{{.HTTPBody | regexFindAllSubmatch "p([a-z]+)ch" }}"
+          }
```
Comment (zhouzhuojie) [inline demo_templates/regex.yaml]: @ziru added examples

Comment (ziru) [reply demo_templates/regex.yaml]: nit: can we add rendered result as well via comments?  best with the use of `index`.
also, do you have any resource links for golang templates?   mockr users do not need to know golang a ton, but would need to know golang templates.

**Pull request merged**